### PR TITLE
<ButtonGroup />

### LINKED ISSUE #46: Implement <ButtonGroup /> component
Image

Image

Please note that this component is responsive (see Figma for the details).

### PR DESCRIPTION
**Pull request by vitalii-ilchenko-ssa-group**
Closes #46

### CODE REVIEW

File: packages/core/src/components/ButtonGroup/ButtonGroup.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { Meta, StoryObj } from '@storybook/react';
+import { css, useTheme } from '@emotion/react';
+import { ButtonGroup } from './ButtonGroup';
+
+const items = [
```
Comment (react-developer-ssa-group) [inline packages/core/src/components/ButtonGroup/ButtonGroup.stories.tsx:5]: I think, we could move out items const to the external file? We use it twice

File: packages/core/src/components/ButtonGroup/ButtonGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react';
+import { Theme, css } from '@emotion/react';
+import Button from '@components/Button';
+import Typography from '@components/Typography';
+import { ButtonGroupProps, Item } from './types';
+
+const ButtonItem = (theme: Theme) => css`
```
Comment (arthur-kashaiev-ssa-group) [inline packages/core/src/components/ButtonGroup/ButtonGroup.tsx:7]: Let's move it to a separate file - `styles.ts`

File: packages/core/src/components/ButtonGroup/ButtonGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  &.active {
+    background: ${theme.colors.greyFocused};
+  }
+`;
+
+export const ButtonGroup = ({
+  items,
+  cssStyles,
```
Comment (arthur-kashaiev-ssa-group) [inline packages/core/src/components/ButtonGroup/ButtonGroup.tsx:39]: I don't like this name. We don't need to mention this is a `CSS` style. It's obvious. It's better to show that this style relates to the button component.

File: packages/core/src/components/ButtonGroup/types.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { SerializedStyles } from '@emotion/react';
+
+export type Item = {
```
Comment (arthur-kashaiev-ssa-group) [inline packages/core/src/components/ButtonGroup/types.ts:3]: Let's rename this to ButtonGroupItem. 
If we decide to export this type, we'll likely have a duplicate name (`Item`).

File: packages/core/src/components/ButtonGroup/ButtonGroup.tsx (hunk 2 above)
Comment (arthur-kashaiev-ssa-group) [inline packages/core/src/components/ButtonGroup/ButtonGroup.tsx:7]: Did you take into account responsive styles?

**Review by arthur-kashaiev-ssa-group — CHANGES_REQUESTED** [review]

File: packages/core/src/components/ButtonGroup/ButtonGroup.stories.tsx (hunk 1 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/core/src/components/ButtonGroup/ButtonGroup.stories.tsx:5]: done

File: packages/core/src/components/ButtonGroup/ButtonGroup.tsx (hunk 2 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/core/src/components/ButtonGroup/ButtonGroup.tsx:7]: done

File: packages/core/src/components/ButtonGroup/ButtonGroup.tsx (hunk 3 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/core/src/components/ButtonGroup/ButtonGroup.tsx:39]: done

File: packages/core/src/components/ButtonGroup/types.ts (hunk 4 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/core/src/components/ButtonGroup/types.ts:3]: Done

File: packages/core/src/components/ButtonGroup/ButtonGroup.tsx (hunk 2 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/core/src/components/ButtonGroup/ButtonGroup.tsx:7]: Done

File: packages/core/src/components/ButtonGroup/styles.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+  &.active {
+    background: ${theme.colors.greyFocused};
+  }
+
+  ${theme.mediaQueries.md} {
+    min-width: 65px;
+    padding: 12px;
+
+    * {
```
Comment (arthur-kashaiev-ssa-group) [inline packages/core/src/components/ButtonGroup/styles.ts:40]: Are you sure about this selector? This is going to affect all the HTML elements on the page if I'm not mistaken

**Review by arthur-kashaiev-ssa-group — CHANGES_REQUESTED** [review]

File: packages/core/src/components/ButtonGroup/styles.ts (hunk 5 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/core/src/components/ButtonGroup/styles.ts]: Changed on 'p'.

**Review by arthur-kashaiev-ssa-group — APPROVED** [review]

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 5  | ❗       |
    


*❗ custom/widgets-meal-nutrients--hint-opened__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ custom/widgets-meal-nutrients--hint-opened__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/widgets-heart-rate--default__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/widgets-heart-rate--default__[w390px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/widgets-meal-nutrients--default__[w900px]*: 


| Before | After |
| --- | --- |
|  |  |

**Review by react-developer-ssa-group — APPROVED** [review]

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 8       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 2  | ❗       |
    


<details>
<summary>🟡 storybook/components-button-group--custom-style__[w1440px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/components-button-group--custom-style__[w1920px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/components-button-group--custom-style__[w390px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/components-button-group--custom-style__[w900px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/components-button-group--default__[w1440px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/components-button-group--default__[w1920px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/components-button-group--default__[w390px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



<details>
<summary>🟡 storybook/components-button-group--default__[w900px] (click to expand)</summary>

| Image |
| --- |
|  |
</details>



*❗ storybook/utility-responsive-image--responsive-image__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/utility-responsive-image--responsive-image__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 7  | ❗       |
    


<details>
<summary>❗ custom/widgets-linkstabbar--current-focus__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-meal-nutrients--hint-opened__[w1440px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-meal-nutrients--hint-opened__[w390px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-navbar-opened__[w1440px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-heart-rate--default__[w1440px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-heart-rate--default__[w390px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-meal-nutrients--default__[w900px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>

**Pull request merged**